### ISSUE TITLE
When 'subscriber' create a new group document, the document gets not assigned to the group

### ISSUE THREAD

**Issue by cherbst** [boonebgorges/buddypress-docs#482]
With git bisect, I was able to track this down to the following commit:
6e3b3c25c3d5fbe36a6a4e35a500b3319024067a

It seems like after this commit, in the template `includes/templates/docs/single/edit.php`, the test for
`current_user_can( 'bp_docs_manage' )` fails and thus the whole `doc-associated-group` block does not get included into the template. This leads to the associated group not being set on the doc.

**Comment by dcavins** [comment]
Hi @cherbst-

Thanks for the report. It looks like the problem was introduced in a472dec1f3b6. 

The new caps handler is getting a dummy doc on the create page using `bp_docs_get_doc_for_caps()` that is a theme compat-style dummy:
```
WP_Post Object
(
    [ID] => 0
    [post_author] => 2
    [post_date] => 0
    [post_date_gmt] => 0
    [post_content] => 
    [post_title] => Create a Doc
    [post_excerpt] => 
    [post_status] => publish
    [comment_status] => closed
    [ping_status] => 
    [post_password] => 
    [post_name] => 
    [to_ping] => 
    [pinged] => 
    [post_modified] => 0
    [post_modified_gmt] => 0
    [post_content_filtered] => 
    [post_parent] => 0
    [guid] => 
    [menu_order] => 0
    [post_type] => bp_doc
    [post_mime_type] => 
    [comment_count] => 0
    [filter] => raw
    [is_404] => 
    [is_page] => 
    [is_single] => 
    [is_archive] => 1
    [is_tax] => 
)
```

By setting the `post_author` of the "create a post" dummy post, I can make the caps work as expected (then the switch drops into the 'creator' case and is allowed to proceed), but I'm not sure if that would have other negative consequences. See https://github.com/dcavins/buddypress-docs/commit/7d37e323cc998a674a1ef64978a8f01b4eed62b6.

@cherbst would you please check that this change fixes the problem on your installation and report back? Thanks!

**Comment by cherbst** [comment]
I can confirm that dcavins@7d37e32 fixes the problem for me.

**Comment by dcavins** [comment]
Thanks for your feedback. I'll check with @boonebgorges to see whether he can think of what this change might break. :)

**Comment by boonebgorges** [comment]
This fix seems good to me. I can't think of any obvious breakages.

**Comment by dcavins** [comment]
Shall I apply the changes to 1.8.x or 1.9.x or both?

**Comment by boonebgorges** [comment]
Both, if it's easy :)

**Comment by dcavins** [comment]
It so is. :)

**Comment by dcavins** [comment]
Fixed in 1.9.x: a10f13d60d287 and in 1.8.x: f20958b2b4.

Thanks for the report, @cherbst!
